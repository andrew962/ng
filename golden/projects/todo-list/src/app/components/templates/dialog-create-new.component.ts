import { Component, OnInit } from '@angular/core';
import { FormBuilder, FormGroup, Validators } from '@angular/forms';
import { uid } from "uid";
import { NewItemList } from '../../models/newitem.model';
import { MatDialog, MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';

@Component({
  selector: 'app-dialog-create-new',
  templateUrl: './dialog-create-new.component.html',
  styles: [
  ]
})
export class DialogCreateNewComponent implements OnInit {
  itemFG: FormGroup;
  constructor(fb: FormBuilder,
    private dialogRef: MatDialogRef<DialogCreateNewComponent, NewItemList>) {
    this.itemFG = fb.group({
      Title: [, Validators.required],
      Content: []
    })
  }

  ngOnInit(): void {
    this.itemFG.markAllAsTouched();
  }
  btnAdd() {
    let newItem: NewItemList = {
      Uid: uid(16),
      Title: this.itemFG.controls.Title.value,
      Content: this.itemFG.controls.Content.value
    }
    this.dialogRef.close(newItem)
    // alert(JSON.stringify(newItem));
  }
  get disabledBtn(): boolean {
    return this.itemFG.invalid;
  }

}
