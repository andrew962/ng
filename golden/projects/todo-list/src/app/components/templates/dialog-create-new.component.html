<h2 mat-dialog-title>Nota</h2>
<mat-dialog-content class="mat-typography">
    <form [formGroup]="itemFG">
        <div class="row">
            <div class="col-12">
                <mat-form-field class="w-100">
                    <mat-label>Título</mat-label>
                    <input matInput formControlName="Title">
                    <mat-error>El título es requierido</mat-error>
                </mat-form-field>
            </div>
            <div class="col-12">
                <mat-form-field class="w-100">
                    <mat-label>Contenido</mat-label>
                    <textarea matInput formControlName="Content"></textarea>
                </mat-form-field>
            </div>
        </div>
    </form>
</mat-dialog-content>
<mat-dialog-actions class="justify-content-end">
    <button class="btn btn-sm" mat-dialog-close>Cancelar</button>
    <button class="btn btn-success btn-sm" [disabled]="disabledBtn" (click)="btnAdd()" [mat-dialog-close]="true" cdkFocusInitial>
    <i class="fas fa-plus"></i>
    Agregar
  </button>
</mat-dialog-actions>